crates/wit-component/src/encoding/world.rs

```diff
@@ -414,22 +414,27 @@ impl<'a> ComponentWorld<'a> {
 
 #[derive(Default)]
 struct Required<'a> {
-    interface_funcs: IndexMap<Option<InterfaceId>, IndexSet<&'a str>>,
+    interface_funcs: IndexMap<Option<InterfaceId>, IndexSet<(&'a str, AbiVariant)>>,
     resource_drops: IndexSet<TypeId>,
 }
 
 impl ImportedInterface {
     fn add_func(&mut self, required: &Required<'_>, resolve: &Resolve, func: &Function) {
-        match required.interface_funcs.get(&self.interface) {
-            Some(set) if set.contains(func.name.as_str()) => {}
+        let abi = match required.interface_funcs.get(&self.interface) {
+            Some(set) if set.contains(&(func.name.as_str(), AbiVariant::GuestImport)) => {
+                AbiVariant::GuestImport
+            }
+            Some(set) if set.contains(&(func.name.as_str(), AbiVariant::GuestImportAsync)) => {
+                AbiVariant::GuestImportAsync
+            }
```

Comment:
This looks like it might not handle the case that a function is imported both sync and async?

Reply:
Good catch.  I've just added a test for that, and it is indeed failing.  Will work on fixing it.

Reply:
Fixed!